snippets/spring-data/query.json

```diff
@@ -0,0 +1,32 @@
+{
+    "Create SQL++ Query": {
+        "prefix": ["@cbsdq"],
+        "body": ["public static N1QLQuery buildQuery(N1QLExpression expression, QueryScanConsistency scanConsistency) {","    QueryOptions opts = QueryOptions.queryOptions().scanConsistency(scanConsistency);","","    return new N1QLQuery(expression, opts);","}","","N1QLQuery query = buildQuery(${1:expression}, ${2:scanConsistency});","${3:CouchbaseTemplate}.findByQuery(${4:Model}.class).matching(query);"],
```

Comment:
Why does this one have part of its logic inside a method and the others don't?

Reply:
The examples with N1QLQuery are problematic - as there are no tests for these. I'm was actually surprised that these still exist.  And when I tried them out they didn't work correctly.   The values from 'opts' are not used.